### ISSUE TITLE
rake cucumber is failed on Ruby 2.4

### ISSUE THREAD

**Issue by junaruga** [cucumber/cucumber-ruby#1070]
## Summary

`bundle exec rake cucumber` is failed on Ruby 2.4.

## Expected Behavior

`bundle exec rake cucumber` is done successfully.

## Current Behavior

`bundle exec rake cucumber` is failed.
`bundle exec rake spec` is done successfully.

## Possible Solution

Fix tests

## Steps to Reproduce (for bugs)

Patch this at first.
https://github.com/cucumber/cucumber-ruby/pull/1069

Then

```
$ bundle install --path vendor/bundle
```

```
$ bundle exec rake cucumber
...
Failing Scenarios:
cucumber features/docs/cli/dry_run.feature:50 # Scenario: In strict mode with an undefined step
cucumber features/docs/cli/showing_differences.feature:8 # Scenario: Run single failing scenario with default diff enabled 
cucumber features/docs/cli/strict_mode.feature:20 # Scenario: Fail with --strict due to undefined step
cucumber features/docs/cli/strict_mode.feature:35 # Scenario: Fail with --strict due to pending step
cucumber features/docs/defining_steps/nested_steps.feature:131 # Scenario: Undefined nested step
cucumber features/docs/defining_steps/table_diffing.feature:7 # Scenario: Extra row
cucumber features/docs/exception_in_after_step_hook.feature:24 # Scenario: Handle Exception in standard scenario step and carry on
cucumber features/docs/exception_in_after_step_hook.feature:57 # Scenario: Handle Exception in scenario outline table row and carry on
cucumber features/docs/exception_in_around_hook.feature:15 # Scenario: Exception before the test case is run
cucumber features/docs/exception_in_around_hook.feature:47 # Scenario: Exception after the test case is run
cucumber features/docs/exception_in_before_hook.feature:44 # Scenario: Handle Exception in Before hook for Scenario with Background
cucumber features/docs/formatters/json_formatter.feature:289 # Scenario: DocString
cucumber features/docs/formatters/junit_formatter.feature:178 # Scenario: pending and undefined steps with strict option should fail
cucumber features/docs/formatters/junit_formatter.feature:332 # Scenario: strict mode with --expand option, one feature, one scenario outline, four examples: one passing, one failing, one pending, one undefined
cucumber features/docs/formatters/pretty_formatter.feature:31 # Scenario: Hook output should be printed before hook exception
cucumber features/docs/getting_started.feature:7 # Scenario: Run Cucumber in an empty directory
cucumber features/docs/gherkin/background.feature:363 # Scenario: run a feature with a background that is pending
cucumber features/docs/writing_support_code/tagged_hooks.feature:34 # Scenario: omit tagged hook
cucumber features/docs/writing_support_code/tagged_hooks.feature:69 # Scenario: Omit example hook

199 scenarios (19 failed, 180 passed)
1177 steps (19 failed, 1 skipped, 1157 passed)
```

## Context & Motivation

We are happy to run cucumber on Ruby 2.4.

## Your Environment

* Version used:

```
$ ruby -v
ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-linux]
```

```
$ bundle -v
Bundler version 1.13.7
```

* Operating System and version:

Fedora 24

* Link to your project:
  https://admin.fedoraproject.org/pkgdb/package/rpms/rubygem-cucumber/

**Comment by mattwynne** [comment]
Thanks for investigating @junaruga. First step here would be to open a WIP PR with the 2.4 build added to travis, so we can all see and talk about the failures. I haven't used 2.4 myself yet but I'm eager for us to support it ASAP.

**Comment by junaruga** [comment]
@mattwynne ok I will send PR adding Ruby 2.4.0 after #1069 is merged. thanks!

**Comment by junaruga** [comment]
@mattwynne ok. I created PR for that now, as I have heard json-1.8.6 (Ruby 2.4.0 compatibility) was released. thanks!
https://github.com/cucumber/cucumber-ruby/pull/1075

**Comment by junaruga** [comment]
I think my solution for failure tests by `simplecov` warnings, was not good.
I would close this PR.
Thanks for taking time for this PR.

**Comment by junaruga** [comment]
It is fixed by #1079 #1080
